trin-core/src/types/validation.rs

```diff
@@ -13,9 +13,9 @@ use crate::{
     utils::provider::TrustedProvider,
 };
 
-pub const MERGE_BLOCK_NUMBER: u64 = 15_537_394u64;
+pub const MERGE_BLOCK_NUMBER: u64 = 15_537_393u64;
```

Comment:
This change (and the other parts where the accumulator ends one header earlier) seems like it belongs in its own commit separate from the fix to the SSZ structure.